### ISSUE TITLE
Sorry to bother you again，I have some questions about the paper

### ISSUE THREAD

**Issue by Zyazz** [zhuchen03/VIBNet#4]
Hi ，
I have some questions about the paper.
In the paper, alpha is approximately equal to 0. In the code, alpha is greater than a threshold. How should the two be understood? In addition, how is this threshold selected and what is the reference basis.

**Comment by daib13** [comment]
Hi @Zyazz , theoretically speaking, \alpha=0 corresponds to the fact that the neuron/channel encodes no information from its previous layer. However, in practice, it is impossible for \alpha to be exactly zero since VIBNet is trained stochastically. So in practice, we set a threshold for \alpha (or equivalently \log\alpha). This is explained on page 12 of the arxiv version.

The reason that we can prune the neurons/channels with very small but not exactly zero \alpha is that the corresponding neurons/channels encode too little (negligible) information such that pruning them will not harm the performance/accuracy.

In our experiments, we also noticed that the performance with respect to both accuracy and compression was insensitive to the choice of the threshold since there is generally a clear separation between redundant and informative neurons. You can also refer to " Bayesian compression for deep learning" which used the same threshold as ours. Of course you can try other thresholds which may lead to slightly different results.

**Comment by Zyazz** [comment]
thanks ！

**Comment by Zyazz** [comment]
For a CNN, it is usually not to add IB  to the input layer. right？

**Comment by daib13** [comment]
@Zyazz right. There are only 3 channels for RGB images. Pruning any channel is not reasonable.

**Comment by Zyazz** [comment]
@daib13  In lenet5-caffe，network structure is   
conv(20)→ ib(20)→ conv(50)→ ib(50) →flatten(800)→ ib(800) →fc(500) →ib(500)  →out (10) , right? 
or in VIBnet, there is no ib(800)? because in " Bayesian compression for deep learning",the structure is 20→ 50 →800→ 500→10

**Comment by DYJ1111** [comment]
Hi, have you met that the loss and kld are nan? I found the logalpha in the code is nan? 

    def get_logalpha(self):
        print('get logalpha', self.post_z_logD.data - torch.log(self.post_z_mu.data.pow(2) + self.epsilon))
        return self.post_z_logD.data - torch.log(self.post_z_mu.data.pow(2) + self.epsilon)

the code above return is nan.
